File: d66-social/src/main/java/com/eroldmr/d66/subreddit/subreddit/Subreddit.java
package com.eroldmr.d66.subreddit.subreddit;

import com.eroldmr.d66.appuser.AppUser;
import com.eroldmr.d66.subreddit.post.Post;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

import java.time.Instant;
import java.util.List;

import static javax.persistence.FetchType.LAZY;

/**
 * @author Mompati 'Patco' Keetile
 * @created 19-12-2022 @ 11:53
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Subreddit {
  @Id
  @SequenceGenerator(name = "subreddit_id_seq", sequenceName = "subreddit_id_seq", allocationSize = 1)
  @GeneratedValue(generator = "subreddit_id_seq", strategy = GenerationType.IDENTITY)
  private Long id;
  @NotBlank(message = "Community name is required.")
  private String name;
  @NotBlank(message = "Description is required.")
  private String description;
  @OneToMany(fetch = LAZY)
  private List<Post> posts;
  private Instant createOn;
  @ManyToOne(fetch = LAZY)
  private AppUser appUser;
}

File: d66-social/src/main/kotlin/com/eroldme/d66/subredit/subredit/Subreddit.kt
package com.eroldme.d66.subredit.subredit

import com.eroldme.d66.subredit.post.Post
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor
import java.time.Instant
import javax.persistence.*
import javax.persistence.FetchType.LAZY
import javax.validation.constraints.NotBlank

/**
 * @author Mompati 'Patco' Keetile
 * @since 06-12-2022 @ 14:12
 *
 */
@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
class Subreddit(
    @Id
    @SequenceGenerator(name = "subreddit_id", sequenceName = "post_id", allocationSize = 1)
    @GeneratedValue(generator = "subreddit_id", strategy = GenerationType.IDENTITY)
    private val id: Long,
    @NotBlank(message = "Community name is required.")
    private val name: String,
    @NotBlank(message = "Description name is required.")
    private val description: String,
    @OneToMany( fetch = LAZY)
    private val posts: List<Post>,
    private val createdDate: Instant,
//    TODO: @ManyToOne(fetch = LAZY) private val user: User
)